### ISSUE TITLE
Custom type parsing

### ISSUE THREAD

**Issue by ozydingo** [ozydingo/despecable#1]
Suggested call signature:

```
datetime :create_after, ->(val){Chronic.parse(val)}, **opts
```

**Comment by ozydingo** [comment]
We can call some validations around the Proc's call -- can we / should we / how do we validate that the proc returned a reasonable value? Can we make it easy for this use case to specify error conditions / handling / messages?

**Comment by ozydingo** [comment]
Released 0.3.0 using syntax:

```
datetime :create_after do |name, val, opts|
  Chronic.parse(val)
end
```
